tsfile/src/main/java/org/apache/iotdb/tsfile/write/writer/TsFileIOWriter.java

```diff
@@ -129,54 +131,53 @@ protected void startFile() throws IOException {
    * @param deviceId device id
    */
   public void startChunkGroup(String deviceId) throws IOException {
+    this.deviceId = deviceId;
+    currentChunkGroupStartOffset = out.getPosition();
     logger.debug("start chunk group:{}, file position {}", deviceId, out.getPosition());
-    currentChunkGroupMetaData = new ChunkGroupMetaData(deviceId, new ArrayList<>(),
-        out.getPosition());
+    chunkMetaDataList = new ArrayList<>();
   }
 
   /**
    * end chunk and write some log.
    * If there is no data in the chunk group, nothing will be flushed.
    */
   public void endChunkGroup(long version) throws IOException {
```

Comment:
If `version` is useless, please remove it.

Reply:
Since ``version`` is used in merge part, I didn't remove it. I'll consider change this when IoTDB server works properly.